### PR TITLE
docs: cardinal API reference

### PR DESCRIPTION
**Pull request by technicallyty**
## What is the purpose of the change

adds API reference for Cardinal. There is A LOT to add in Cardinal, so this can be thought of as a skeleton for now - to get the ball rolling.

## Brief Changelog

n/a

## Testing and Verifying

n/a

### CODE REVIEW

File: docs/pages/Cardinal/CoreConcepts/Components.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+```go
+err := HealthComponent.Set(world, id, Health{Amount: 50, Cap: 100})
+```
+
+## Updating Component Data
+Aside from directly setting component data (effectively a full overwrite), component types provide a method `Update`, which updates component data with a more ergonomic callback pattern. `Update` takes a pointer to `World`, the ID of the entity to update, and a callback function. The
+callback function supplies the component struct directly, and expects a return value of the updated component struct. `Update` may return an error.
+
+```
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Components.mdx:44]: Missing "go" on the code comment

File: docs/pages/Cardinal/CoreConcepts/Components.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+err := HealthComponent.Update(world, id, func(h Health) Health {
+    // take away 10 from the total heatlh
+    h.Amount -= 10
+    return h
+})
+```
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Components.mdx:50]: Optional: What do you think about including something like:

The `Update` pattern above is equivalent to:

```go
    h, err := HealthComponent.Get(world, id)
    if err != nil {
        return err
    }
    h.Amount -= 10
    err = HealthComponent.Set(world, id, h)
    if err != nil {
        return err
    }
```

File: docs/pages/Cardinal/CoreConcepts/Components.mdx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+## Entity Iteration
+Components can also iterate over all entities that contain this component with the method `Each`. `Each` takes a pointer to world, and a callback function which will supplies each entity ID at every call. The callback function expects a boolean value to be returned, which indicates whether the
+iteration should continue. A return value of `true` indicates the iteration should continue, while `false` indicates the iteration should stop. `Each` may return an error.
+
+```go
+err := HealthComponent.Each(world, func(id storage.EntityID) bool {
+    // do something with entity ID...
+	return true // continue the iteration
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Components.mdx:59]: indent mismatch

File: docs/pages/Cardinal/CoreConcepts/Reads.mdx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+type PlayerReply struct {
+    Items       []Item
+    Position    Coordinates
+}
+```
+
+Once the request and reply structs have been defined, instantiate the Read. Note that the `NewReadType` function takes a name argument of type string. Cardinal uses this name to automatically create the endpoint for this read. For example, passing "foo-bar" would result in the endpoint
+`address:port/read/foo-bar`. (TODO: update this after we finalize how we want to handle endpoints)
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Reads.mdx:22]: Add "CAR-139" to this comment so we can later search for it.

https://linear.app/arguslabs/issue/CAR-139/refactor-http-endpoints-to-follow-a-clear-readtx-pattern

File: docs/pages/Cardinal/CoreConcepts/Reads.mdx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+The `NewReadType` function accepts Options to alter the read.
+
+### WithReadEVMSupport
+
+import { Callout } from 'nextra/components'
+
+import Link from 'next/link'
+
+<Callout type={"warning"}>
+    Not all Go types are supported when using `WithReadEVMSupport`. <Link href="/cardinal/CoreConcepts/evm-support">Read More</Link>
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Reads.mdx:48]: Is this case sensitive? I think Cardinal needs to be upper case.

File: docs/pages/Cardinal/CoreConcepts/Transactions.mdx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+type PickUpItemOutput struct {
+	Success bool
+}
+```
+
+Then create the Cardinal transaction type using the `NewTransactionType` function in the ecs package. Note, this
+function takes a `name` argument. The name passed into the transaction gets used as the endpoint on the game server
+to execute this transaction. For example, if the transaction name passed in was "foo-bar", the resulting endpoint would
+be `address:port/tx/foo-bar` (TODO: update this after we finalize how we want to handle endpoints)
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Transactions.mdx:19]: Same as above: Add "CAR-139" to the comment text.

File: docs/pages/Cardinal/CoreConcepts/Transactions.mdx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+```go
+import "github.com/argus-labs/world-engine/cardinal/ecs"
+
+
+var PickUpTransaction = ecs.NewTransactionType[PickUpItemInput, PickupItemOutput]("pick-up-item")
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Transactions.mdx:25]: nit: Pickup is one word.

File: docs/pages/Cardinal/CoreConcepts/Transactions.mdx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+The `NewTransactionType` function accepts Options to alter the transaction.
+
+### WithTxEVMSupport
+
+import { Callout } from 'nextra/components'
+
+import Link from 'next/link'
+
+<Callout type={"warning"}>
+   Not all Go types are supported when using `WithTxEVMSupport`. <Link href="/cardinal/CoreConcepts/evm-support">Read More</Link>
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/Transactions.mdx:39]: Same a above: Cardinal should be upper case.

File: docs/pages/Cardinal/CoreConcepts/World.mdx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+```go
+world.AddSystem(system.Movement)
+world.AddSystem(system.Attack)
+world.AddSystem(system.Physics)
+```
+
+## Loading Game State
+
+Game state, if any, must be loaded before starting the game loop.
```
Comment (jerargus) [inline docs/pages/Cardinal/CoreConcepts/World.mdx:92]: How about adding a note that RegisterReads, RegisterComponents, and RegisterTransactions can be called at most once, and LoadGameState must be called after Register* functions and all AddSystem calls.

**Review by jerargus — APPROVED** [review]

File: cardinal/ecs/abi.go [hunk 10]
```diff
@@ -42,8 +42,8 @@ func GenerateABIType(goStruct any) (*abi.Type, error) {
 }
 
 func goTypeToSolidityType(t string, tag string) (string, error) {
-	// first handle the most special type. []bytes. this is very specific for ethereum, in that it translates to 'bytes'
-	if t == "[]bytes" {
```
Comment (smsunarto) [inline cardinal/ecs/abi.go:46]: is this intended to be here?

File: docs/pages/Cardinal/_meta.json [hunk 11]
```diff
@@ -0,0 +1,5 @@
+{
+  "CoreConcepts": {
```
Comment (smsunarto) [inline docs/pages/Cardinal/_meta.json:2]: I don't think "Core Concepts" is apt here. Since this part of the docs is more of an API reference, we should call it "API Reference instead

File: docs/pages/Cardinal/CoreConcepts/World.mdx [hunk 12]
```diff
@@ -0,0 +1,141 @@
+# World
```
Comment (smsunarto) [inline docs/pages/Cardinal/CoreConcepts/World.mdx:1]: In general, I think for Components/Reads/Systems/Transactions/World we should follow the docs pattern that https://nextjs.org/docs/app/api-reference/components/image has

1. Provide a full example of how to define/use it
2. Provide a table with the parameters, types, and description

File: docs/pages/Cardinal/CoreConcepts/Systems.mdx [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+Systems give you a reference to the world, as well as the transaction queue. This allows you to access any transaction currently queued for this tick, as well as update any component state.
+
+## Getting Transactions
+
+In order to get specific transactions from the queue, utilize the transaction's `In` method. This transaction method takes the queue as an argument, and returns a slice of `TxData`, which contains data related to that transaction.
+
+```go
+import "github.com/argus-labs/world-engine/cardinal/ecs"
+
+func SomeSystem(world *ecs.World, queue *ecs.TransactionQueue) error {
```
Comment (smsunarto) [inline docs/pages/Cardinal/CoreConcepts/Systems.mdx:20]: I prefer examples that have more concrete naming so that it provides better illustration to the reader. Instead of `SomeSystem` it should be `AttackSystem` and instead of `SomeTx` it should be `AttackTx`

File: cardinal/ecs/abi.go (hunk 10 above)
Comment (technicallyty) [reply cardinal/ecs/abi.go:46]: yep, small stealth fix

**Comment by technicallyty** [comment]
quite a lot of docs in here, so i expect there to be typos and brain farts 🤡 pls help catch them

File: docs/pages/Cardinal/API-Reference/Transactions.mdx [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+
+This API reference will help you understand how to use transactions and configuration options available for transactions.
+
+```go
+package transactions
+
+import "pkg.world.dev/world-engine/cardinal/ecs"
+
+// AttackInput is the input for the Attack transaction.
+type AttackInput struct {
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/Transactions.mdx:11]: Can we call this `AttackMsg`

File: docs/pages/Cardinal/API-Reference/Transactions.mdx [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+    TargetID uint64
+}
+
+// AttackOutput is the output for the Attack transaction.
+type AttackOutput struct {
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/Transactions.mdx:16]: Can we call this `AttackReceipt`

File: docs/pages/Cardinal/API-Reference/Transactions.mdx [hunk 16]
```diff
@@ -0,0 +1,208 @@
+# Transactions
+
+This API reference will help you understand how to use transactions and configuration options available for transactions.
+
+```go
+package transactions
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/Transactions.mdx:6]: i think we called this package `tx` in the starter-game-template. let's try to make it consistent to reduce confusion

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 15 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:16]: `Receipts` are already defined as:

```go
// Receipt contains a transaction ID, an arbitrary result, and a list of errors.
type Receipt struct {
	ID     transaction.TxID
	Result any
	Errs   []error
}
```

I think naming this `AttackReceipt` might be confusing. Additionally, the signature for `NewTransactionType`'s type parameters is `In, Out`, so i used `Output` to stay consistent there.

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 17]
```diff
@@ -0,0 +1,529 @@
...
+| Type  | Description                                      |
+|--------------|--------------------------------------------------|
+| *World       | A pointer to the newly created World instance.  |
+| error        | An error indicating any issues during creation. |
+
+### Options
+
+#### WithAdapter
+
+The `WithAdapater` world option enables `World` objects to read transactions from the EVM base shard, and rebuild game state from those transactions.
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/World.mdx:59]: I think we can make this more

The `WithAdapter` option allows Cardinal to post and recover transaction calldata from the sequencer (EVM base shard). When running Cardinal as a rollup in production, `WithAdapter` must be used.

Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/World.mdx:59]: Also, typo. `WithAdapater` -> `WithAdapter`

Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx:59]: This option does not enable the world to post any data, only read. Posting is the job of the server

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 16 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: [docs: rename example package from transactions to tx](https://github.com/Argus-Labs/world-engine/pull/200/commits/f495c8ad1f9bb7a9d162ce892d556a442b5bd176)

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 18]
```diff
@@ -0,0 +1,529 @@
...
+import (
+    "pkg.world.dev/world-engine/cardinal/ecs"
+
+    "github.com/my-username/my-world-engine-project/systems"
+)
+
+func main() {
+    // ... world setup ...
+
+    world.AddSystems(systems.MoveSystem, systems.HealthRegenSystem, systems.AttackSystem)
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/World.mdx:118]: Add a comment here that mention that MoveSystem will run first, followed HealthRegenSystem, etc ...

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 17 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx]: [docs: add note that WithAdapter must be used in prod](https://github.com/Argus-Labs/world-engine/pull/200/commits/b9cd972f478ec82a461969aebef779775f6da269) added note for prod

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 19]
```diff
@@ -0,0 +1,529 @@
...
+    "pkg.world.dev/world-engine/cardinal/ecs"
+
+    "github.com/my-username/my-world-engine-project/components"
+)
+
+func main() {
+    // ... world setup ...
+
+    err := world.RegisterComponents(
+		components.LocationComponent,
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/World.mdx:159]: package name in starter-game-template is singular `component` would like to keep it consistent

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 20]
```diff
@@ -0,0 +1,529 @@
...
+    "pkg.world.dev/world-engine/cardinal/ecs"
+
+    "github.com/my-username/my-world-engine-project/reads"
+)
+
+func main() {
+    // ... world setup ...
+
+    err := world.RegisterReads(
+		reads.ReadLocation,
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/World.mdx:206]: package name in starter-game-template is singular `read` would like to keep it consistent

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 21]
```diff
@@ -0,0 +1,529 @@
...
+    "pkg.world.dev/world-engine/cardinal/ecs"
+
+    "github.com/my-username/my-world-engine-project/txs"
+)
+
+func main() {
+    // ... world setup ...
+
+    err := world.RegisterTransactions(
+		txs.MoveTx,
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/World.mdx:254]: package name in starter-game-template is singular `tx` would like to keep it consistent

File: docs/pages/Cardinal/API-Reference/evm-support.mdx [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+- string
+- bool
+- int8 - int64
+- uint8 - uint64
+- github.com/ethereum/go-ethereum/common.Address (resolves to `address` in ABI encoding)
+- *math/big.Int (resolves to a specified int or uint above 64 bits i.e. uint128)
+
+Additionally, slices of all the above types work as well.
+
+import { Callout } from 'nextra/components'
```
Comment (smsunarto) [inline docs/pages/Cardinal/API-Reference/evm-support.mdx:19]: Can we move this import the top

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 21 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx]: [package consistency:](https://github.com/Argus-Labs/world-engine/pull/200/commits/3ace678c1ac3dfe79a376c766e89708b80e22404)

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 20 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx]: [package consistency:](https://github.com/Argus-Labs/world-engine/pull/200/commits/3ace678c1ac3dfe79a376c766e89708b80e22404)

File: docs/pages/cardinal/ecs.mdx [hunk 23]
```diff
@@ -2,7 +2,7 @@
 
 Cardinal follows the ECS (Entity-Component-System) architecture which facilitates fast and efficient updates to game state, as well as enabling painless extensibility to your games.
```
Comment (smsunarto) [inline docs/pages/cardinal/ecs.mdx:3]: follows -> uses

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 19 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx]: [package consistency:](https://github.com/Argus-Labs/world-engine/pull/200/commits/3ace678c1ac3dfe79a376c766e89708b80e22404)

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 18 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx:122]: [docs: system ordering note](https://github.com/Argus-Labs/world-engine/pull/200/commits/9d5e32ba8ee83d3d51ada018e5e543a5788dd2aa)

File: docs/pages/cardinal/ecs.mdx [hunk 24]
```diff
@@ -2,7 +2,7 @@
...
 
-ECS, or Entity-Component-System, stands as a programming paradigm specifically tailored for game development. While ECS itself is conceptually straightforward, its effective implementation demands a nuanced understanding. Familiarizing yourself with the ECS model is crucial before delving into the capabilities of Cardinal.
+ECS stands as a programming paradigm specifically tailored for game development. While ECS itself is conceptually straightforward, its effective implementation demands a nuanced understanding. Familiarizing yourself with the ECS model is crucial before delving into the capabilities of Cardinal.
```
Comment (smsunarto) [inline docs/pages/cardinal/ecs.mdx:5]: > While ECS itself is conceptually straightforward, its effective implementation demands a nuanced understanding. Familiarizing yourself with the ECS model is crucial before delving into the capabilities of Cardinal.

Let's reword this to 

> While ECS is easy to learn, it's important to familiarize yourself with how idiomatic ECS codebase is structured.

File: docs/pages/cardinal/ecs.mdx (hunk 23 above)
Comment (technicallyty) [reply docs/pages/cardinal/ecs.mdx]: [follows -> uses](https://github.com/Argus-Labs/world-engine/pull/200/commits/6b8a57f88a935e7876605fbe009f2de6f34117f8)

File: docs/pages/Cardinal/API-Reference/evm-support.mdx (hunk 22 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/evm-support.mdx]: [move import to top](https://github.com/Argus-Labs/world-engine/pull/200/commits/2a47bf0d481954a256601a39d1a43fb773dc35cf)

File: docs/pages/cardinal/ecs.mdx
Comment (smsunarto) [inline docs/pages/cardinal/ecs.mdx:1]: CleanShot 2023-08-21 at 15 27 59@2x

You might want to fix this

Comment (technicallyty) [reply docs/pages/cardinal/ecs.mdx:1]: weird - its not doing this locally. not sure why theres an extra `cardinal` folder in the diff...

Comment (technicallyty) [reply docs/pages/cardinal/ecs.mdx:1]: lol deleting the `pages` directory and then recommitting it fixed it ?? whack

File: docs/pages/cardinal/ecs.mdx (hunk 24 above)
Comment (technicallyty) [reply docs/pages/cardinal/ecs.mdx:5]: i like the direction you're going, but i think we could take the best of both here and do something like:

> While ECS is conceptually straightforward, gaining familiarity with the structured organization of an idiomatic ECS codebase is essential to fully harness its unique advantages.

thoughts?

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 14 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:11]: that deviates from the named type parameters of the `NewTransactionType` function signature:

```go
func NewTransactionType[In, Out any]
```
if we're okay with the mismatch i can make the change

File: docs/pages/cardinal/ecs.mdx (hunk 24 above)
Comment (smsunarto) [reply docs/pages/cardinal/ecs.mdx:5]: lgtm

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 17 above)
Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/World.mdx]: @technicallyty we should definitely create a ticket on linear to consolidate that into the same thing. i think we had this convo a while back.

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 14 above)
Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:11]: i think we should create a ticket to change that too

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 15 above)
Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:16]: to be clear, output is going to do the `Result` field of the receipt struct right?

if that's the case, i think naming it AttackReceipt makes sense because it goes into the receipt struct

File: docs/pages/Cardinal/API-Reference/World.mdx (hunk 17 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/World.mdx]: its still an open ticket - https://linear.app/arguslabs/issue/CAR-110/have-world-automatically-setup-the-transaction-handler

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 14 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:11]: created https://linear.app/arguslabs/issue/CAR-165/change-type-parameters-names-of-transactiontypes

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 15 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:16]: i think the fact that it goes specifically into a field named `Result` lets itself better to being named `AttackResult` - since `Receipt` for us is more broad (errors & result)

Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:16]: I really want to use AttackReceipt here because I want to be very clear that this is going into the receipt. It's not intuitive from `AttackResult` where that result is going.

Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:16]: I'll go ahead and change it (strong opinion weakly held) - we may want to consider more renamings here though - as we now have situations like this:

```go
AttackTx.SetResult(w, txID, AttackReceipt{Success: true})
```

perhaps we want to change this to method `SetReceipt`? or maybe its fine as is. just want to draw attention to the mismatch there.

Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: yeah, i think we should create a ticket to that. Should be SetReceipt

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 14 above)
Comment (jerargus) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx:11]: I commented on the ticket, but I recommend Msg and Result. I'd like to have Receipt be the combination of the Result and any errors.

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 15 above)
Comment (jerargus) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: I agree with Tyler, in that we should have SetResult on the transaction, and that "result" should go into the "result" field in a struct called "receipt". To me, a "result" is what you get after processing a transaction in some happy path. I haven't seen "receipt" used much in software development, so coming up with our own definition (a result and associated errors) feels right.

Comment (namrapatel) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: That also feels intuitively right here to me too^, @smsunarto wdyt?

**Review by pyrofolium — APPROVED** [review]
lgtm. I feel I would be able to provide more useful feedback if I were using the documentation to build something. That's where I'll have the most useful insight as of right now it's hard to say.

File: docs/pages/Cardinal/API-Reference/Transactions.mdx (hunk 15 above)
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: can we revisit here? cc @smsunarto

Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: Can we call it `SetReceiptResult `

Comment (jerargus) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: Meaning we would also have AddReceiptError? 

I prefer `SetResult` and `AddError`

Comment (smsunarto) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: ok, i'll agree with you guys here

**Comment by technicallyty** [comment]
pls wait to merge. a few changes have been made to main cardinal, and agreements have been met on certain naming. i need to update these things before merging.

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 25]
```diff
@@ -0,0 +1,467 @@
...
+```go
+func (w *World) RegisterReads(reads ...AnyReadType) error
+```
+#### Example
+
+```go
+package main
+
+import (
+	"log"
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/World.mdx:240]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 26]
```diff
@@ -0,0 +1,467 @@
...
+```go
+func (w *World) RegisterTransactions(txs ...AnyTransaction) error
+```
+
+#### Example
+```go
+package main
+
+import (
+	"log"
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/World.mdx:286]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 27]
```diff
@@ -0,0 +1,467 @@
...
+```go
+func (w *World) CreateMany(num int, components ...AnyComponentType) ([]EntityID, error)
+```
+
+#### Example
+```go
+package main
+
+import (
+	"log"
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/World.mdx:328]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 28]
```diff
@@ -0,0 +1,467 @@
...
+```go
+func (w *World) Create(components ...AnyComponentType) (EntityID, error)
+```
+
+#### Example
+```go
+package main
+
+import (
+	"log"
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/World.mdx:375]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/World.mdx [hunk 29]
```diff
@@ -0,0 +1,467 @@
...
+```go
+func (w *World) Remove(id storage.EntityID) error
+```
+
+#### Example
+```go
+package main
+
+import (
+	"log"
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/World.mdx:420]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/Systems.mdx [hunk 30]
```diff
@@ -0,0 +1,86 @@
...
+
+## Getting Transactions
+
+In order to get specific transactions from the queue, utilize a `TransactionType`'s `In` method. This `TransactionType` method takes the queue as an argument, and returns a slice of `TxData`, which contains data related to that transaction.
+
+```go
+import "github.com/argus-labs/world-engine/cardinal/ecs"
+
+func AttackSystem(world *World, queue *TransactionQueue, logger *Logger) error {
+    txs := AttackTransaction.In(queue)
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/Systems.mdx:19]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/Transactions.mdx [hunk 31]
```diff
@@ -0,0 +1,214 @@
...
+
+```go
+package tx
+
+import "pkg.world.dev/world-engine/cardinal"
+
+
+// AttackMsg is the input for the Attack transaction.
+type AttackMsg struct {
+    TargetID uint64
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/Transactions.mdx:17]: whitespace mismatch

File: docs/pages/Cardinal/API-Reference/evm-support.mdx [hunk 32]
```diff
@@ -0,0 +1,30 @@
...
+Additionally, slices of all the above types work as well.
+
+<Callout type={"warning"}>
+    When using `*big.Int`, you MUST use a special "evm" struct tag to indicate which underlying evm integer type you want to resolve to. This is due to go-ethereum using `*big.Int` for any uint or int greater than 64 bits.
+</Callout>
+
+Example:
+```go
+type Foo struct {
+    Num *big.Int `evm:"uint256"`
```
Comment (jerargus) [inline docs/pages/Cardinal/API-Reference/evm-support.mdx:28]: nit: this should be tab

File: docs/pages/Cardinal/API-Reference/Reads.mdx [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+		// do some query stuff...
+		reply, err := doSomeQueryLogic(req)
+		return reply, err
+	},
+)
+```
+
+## NewReadType
+
+`NewReadType` creates a new `ReadType`. `ReadType`s wrap around a request and reply type. The request specifies the data needed for the read request. The reply type represents the data requested. Reads also require a function that implements the read.
```
[the remark this answers is not in the document]
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Reads.mdx:39]: good idea, added that

File: docs/pages/Cardinal/API-Reference/Transactions.mdx [hunk 34]
```diff
@@ -0,0 +1,214 @@
...
+|-------------|----------------------|-----------------------------------------------------|
+| world       | *World               | A pointer to a World instance.  |
+| hash          | TxHash    | The hash of the transaction.    |
+
+#### Return Values
+
+| Type        | Description                                                     |
+|--------------------|-----------------------------------------------------------------|
+| Result              | The transaction result.  |
+| []error               | A slice of errors associated with the transaction.            |
```
[the remark this answers is not in the document]
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: i reworded all these to make it easier to indicate that the transactions are associated with some hash

[the remark this answers is not in the document]
Comment (technicallyty) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: lmk if looks good

[the remark this answers is not in the document]
Comment (jerargus) [reply docs/pages/Cardinal/API-Reference/Transactions.mdx]: lgtm
